### PR TITLE
Use C++ style casts

### PR DESCRIPTION
**Pull request by dschwoerer**
/clang-format

Part of #2025 

Doesn't change anything, just makes it more explicit.

### CODE REVIEW

**Comment by ZedThree** [comment]
Just FYI, the clang-format command did work, but there was nothing to do. Maybe I should add some feedback for that case?

File: src/sys/boutexception.cxx [hunk 1]
```diff
@@ -53,8 +53,8 @@ std::string BoutException::getBacktrace() const {
     void * ptr=trace[i];
     if (dladdr(trace[i],&info)){
       // Additionally, check whether this is the default offset for an executable
-      if (info.dli_fbase != (void*)0x400000)
-        ptr=(void*) ((size_t)trace[i]-(size_t)info.dli_fbase);
+      if (info.dli_fbase != reinterpret_cast<void*>(0x400000))
+        ptr=reinterpret_cast<void*>(reinterpret_cast<size_t>(trace[i])-reinterpret_cast<size_t>(info.dli_fbase)));
```
Comment (ZedThree) [inline src/sys/boutexception.cxx:57]: One too many closing brackets here!

```suggestion
        ptr=reinterpret_cast<void*>(reinterpret_cast<size_t>(trace[i])-reinterpret_cast<size_t>(info.dli_fbase));
```

**Comment by dschwoerer** [comment]
/clang-format

Did the first one work? On the second there are emojis, so that seems to have worked, but the first one not?

Also I think the review with `/clang-format` didn't work :sweat_smile: 
Still think it is a great way to ensure all can use the same c`lang-format` :+1:

**Comment by ZedThree** [comment]
I don't think the first one triggered it. The second one did trigger it but I think the syntax error fooled it, so it errored. The :tada: emoji on the third one indicates it finished successfully, although it says it was from me, rather than the bot!

Glad you think it's useful! I'll see if there's any polish can be done

**Pull request merged**